Content.Server/Procedural/DungeonJob.PostGen.cs

```diff
@@ -558,22 +568,24 @@ private async Task PostGen(DungeonEntrancePostGen gen, Dungeon dungeon, EntityUi
 
                     if (!allExterior.Contains(neighbor) ||
                         takenTiles.Contains(neighbor) ||
-                        !_anchorable.TileFree(grid, neighbor, DungeonSystem.CollisionLayer, DungeonSystem.CollisionMask))
+                        !_anchorable.TileFree(gridUid, grid, neighbor, DungeonSystem.CollisionLayer,
+                            DungeonSystem.CollisionMask))
                     {
                         isValid = false;
                         break;
                     }
 
                     // Also check perpendicular that it is free
-                    foreach (var k in new [] {2, 6})
+                    foreach (var k in new[] { 2, 6 })
```

Comment:
evil sloth allocating whole ass array for this

Reply:
probably gets optimized tbh

Reply:
how to optimize it lol -_-